src/analytics/api/models/bigquery_api.py

```
@@ -674,68 +676,154 @@ def compose_dynamic_query(
                 self.get_averaging_columns(
                     BM_pollutant_mapping,
                     frequency,
                     decimal_places,
                     bam_table_name,
                 )
             )
-
             # TODO Clean up by use using `get_columns` helper method
             if pollutant in {"pm2_5", "pm10"} and data_type.value == "raw":
                 # Add dummy column to fix union column number missmatch.
                 bam_pollutant_columns.append("-1 as " + pollutant)
```

Ruff (0.12.2)

Undefined name `bam_pollutant_columns`

(F821)